clicommand/agent_start.go

```diff
@@ -1182,7 +1182,16 @@ var AgentStartCommand = cli.Command{
 			pool.StartStatusServer(ctx, l, cfg.HealthCheckAddr)
 		}
 
-		return pool.Start(ctx)
+		err = pool.Start(ctx)
+		if errors.Is(err, agent.ErrJobAcquisitionFailure) {
+			// If the agent tried to acquire a job, but it couldn't because the job was already taken, we should exit with a
+			// specific exit code so that the caller can know that this job can't be acquired.
+
+			const acquisitionFailedExitCode = 27 // chosen by fair dice roll
+			return cli.NewExitError(err, acquisitionFailedExitCode)
```

Comment:
I wish I could find a better way to do this, It feels like the job of looking up an error and translating it to an exit code should be simpler.

Reply:
i mean, if the return value of a `cli.Command` is an `ExitCoder` (ie, fulfils `interface { ExitCode() int }`, this would happen automatically, but i actually think that that would be worse than what we have here - it would mean pulling a concern of the CLI into the `agent` package. i think this is fine as is for now, and if we end up checking a whole bunch of error types, we can refactor as necessary

Reply:
100% agreed, that's the conclusion I came to, but it was niggling at me